### PR TITLE
Exporting fakes.NewTestEnv.

### PR DESCRIPTION
**Pull request by ola-rozenfeld**
Refactoring the current rexec_test to use it. Note that tests have not
been removed! They just became more easily packaged into test structs
using the new fake.

### CODE REVIEW

File: go/pkg/command/BUILD.bazel [hunk 1]
```diff
@@ -15,4 +16,9 @@ go_test(
     name = "go_default_test",
     srcs = ["command_test.go"],
     embed = [":go_default_library"],
+    deps = [
+        "@com_github_bazelbuild_remote_apis//build/bazel/remote/execution/v2:go_default_library",
```
[the remark this answers is not in the document]
Comment (ola-rozenfeld) [reply go/pkg/command/BUILD.bazel]: I guess I could export this function from the rexec package as well, but it just doesn't make a lot of sense there, I think. What's the harm of making command depend on RE API directly? Note that it already depends on it transitively through the digest package. So there's no difference for the callers.

File: go/pkg/fakes/server.go [hunk 2]
```diff
@@ -57,3 +70,160 @@ func (s *Server) NewTestClient(ctx context.Context) (*rc.Client, error) {
...
+	Client     *rexec.Client
+	Server     *Server
+	ExecRoot   string
+	GrpcClient *rc.Client
+	t          *testing.T
+}
+
+// NewTestEnv initializes a TestEnv containing a fake server, a client connected to it,
+// and a temporary directory used as execution root for inputs and outputs.
+func NewTestEnv(t *testing.T) *TestEnv {
```
[the remark this answers is not in the document]
Comment (ola-rozenfeld) [reply go/pkg/fakes/server.go]: Oh! Yes, most definitely, thank you! I think in Set as well.

[the remark this answers is not in the document]
Comment (ola-rozenfeld) [reply go/pkg/fakes/server.go]: Oh! I thought it was nicer this way, but I guess you're right, returning makes it more obvious that you're supposed to do it. Sure, done.

File: go/pkg/fakes/server.go [hunk 3]
```diff
@@ -57,3 +70,160 @@ func (s *Server) NewTestClient(ctx context.Context) (*rc.Client, error) {
...
+type ExecutionCacheHit bool
+
+// Apply on true will cause the ActionResult to be returned as a cache hit during fake execution.
+func (c ExecutionCacheHit) Apply(ac *repb.ActionResult, s *Server) {
+	s.Exec.Cached = bool(c)
+}
+
+// Set sets up the fake to return the given result on the given command execution.
+// It is not possible to make the fake result in a LocalErrorResultStatus or an InterruptedResultStatus.
+func (e *TestEnv) Set(cmd *command.Command, opt *command.ExecutionOptions, res *command.Result, opts ...option) (cmdDg, acDg digest.Digest) {
```
[the remark this answers is not in the document]
Comment (ola-rozenfeld) [reply go/pkg/fakes/server.go]: Done.

[the remark this answers is not in the document]
Comment (ola-rozenfeld) [reply go/pkg/fakes/server.go]: Right, by design. I want to make the list of possible options fixed to the ones explicitly defined by this package.

File: go/pkg/command/BUILD.bazel (hunk 1 above)
[the remark this answers is not in the document]
Comment (ramymedhat) [reply go/pkg/command/BUILD.bazel]: Didn't notice that digest depends on RE. Current state sgtm.

**Comment by ola-rozenfeld** [comment]
You said SGTM, but you didn't press the approve button :-)

**Pull request merged**